### DISCUSSION TITLE
MPD Gets stuck When Loading Album Artwork And Scrolling From A-Z

### DISCUSSION THREAD

**Discussion by santhosharulvanan** [MusicPlayerDaemon/MPD#1673] [Q&A]
Hello All,

I recently installed MPD in 64bit Raspbian, the library consists of 10,000 files.
In the android client MAFA/MPDROID when I scroll through the albums from A-Z slowly, MPD at a certain point of time disconnects and does not connect any more.

When i ran the below command:

 watch 'ss -np | grep 192.168.0.2 | grep -v ssh'

There are 5 to 6 CLOSE_WAIT commands with the MPD port listed: 6600
![image](https://user-images.githubusercontent.com/28530911/202672721-dbf89a45-47d8-413b-95dc-8c0177588e93.png)
![image](https://user-images.githubusercontent.com/28530911/202678139-fc7dece5-191a-4599-b483-264d536f4c95.png)

Is there any way to get this resovled, when the close_wait connections appear the MPD Client stops working, it says connection timed out.

Please let me know, if this can be fixed will be grateful for your help and support.
Thank you.

Config File:

music_directory		"/var/lib/mpd/music"

playlist_directory		"/var/lib/mpd/playlists"

db_file			"/var/lib/mpd/tag_cache"

log_file			"/var/log/mpd/mpd.log"

pid_file			"/run/mpd/pid"

state_file			"/var/lib/mpd/state"

sticker_file                   "/var/lib/mpd/sticker.sql"

user				"mpd"

bind_to_address			"any"

decoder {
        plugin                  "hybrid_dsd"
        enabled                 "no"
}
decoder {
        plugin        "wildmidi"
        enabled       "no"
        #config_file "/etc/timidity/timidity.cfg"
}

audio_output {
        type            "alsa"
        name            "Allo Digi One"
        device          "hw:sndallodigione,0"   # optional
        mixer_type      "none"      # optional
        dop             "yes"
        auto_resample   "no"
        auto_channels   "no"
        auto_format     "no"
}

Regards
Santhosh

**Comment by MaxKellermann** [comment]
https://www.musicpd.org/help/

**Reply by santhosharulvanan** [reply]
I have tried the tips in troubleshooting document but nothing seemed to help.
MPD Version: Music Player Daemon 0.22.6 (0.22.6)

There is nothing stored in the log file even with --verbose enabled.
MPD just crashes when there are too many CLOSE-WAIT statements.
MPC status returns MPD timed out as well.

systemctl restart mpd doesnt work and I need to reboot the RPI4 for MPD to work again.

I am not able to diagnose the proper reason for this particular timeout as there are no details being logged in the /var/log/mpd/mpd.log file as well.

**Reply by MaxKellermann** [reply]
Your mistake was ignoring this one:
> make sure you have the latest MPD version (0.23.10)

**Reply by santhosharulvanan** [reply]
Hello Max,

Appreciate your help in supporting me.

I have compiled and installed the latest version of MPD in Raspbian 0.23.10.

![image](https://user-images.githubusercontent.com/28530911/202914165-dfd7b89d-2e92-4dc8-be63-5c4a24525d97.png)
I still have the problem with MPD getting stuck when i browse through the Albums in MAFA MPD Client, this happens because my cover.jpg files are in resolution 1000X1000 or due to some other reason which I am not able to understand why.

Total independent albums with different album art for each of them: 1000

MPD in debugging mode runs the query to fetch the artwork.
![image](https://user-images.githubusercontent.com/28530911/202914239-42abe9db-a8bc-4980-aca4-530c71c5ec66.png)
When you scroll from A-Z or A-S some letter in between it runs too many queries to fetch the artwork and I am not sure if this is the reason it freezes.
But at a certain point of time it gets stuck and there is nothing reported in the log, the client looses connection to the MPD server.
It says connected waiting for MPD for a minute and then it times out.

mpd --kill does not respond, I need to reboot the system to get this working.

Network trace shows too many CLOSE-WAIT connections for socket in MPD port.
![image](https://user-images.githubusercontent.com/28530911/202914497-b8a15ba7-4be8-4696-a84b-5d2cf027d1a1.png)

**Comment by petebueh** [comment]
Hello  santhosharulvanan,

I have the same problem when scrolling in Mafa through the album list.

As a (quick) fix I embedded the covers in my FLAC files. Then I disabled "Embed MPD album art" in Mafa and enabled "Enable MPD embedded album art". This works for me.

I have about 1.800 albums in the MPD database. Most of my cover.jpg-files also have a size of 1000px * 1000px.

Best Regards,
Peter
